networking/GameData.gd

```diff
@@ -52,32 +49,33 @@ func reset():
 	self.players = {}
 
 
-func get_current_player():
+func get_current_player() -> PlayerData:
 	var id := get_tree().get_network_unique_id()
 	if players.has(id):
-		return players[id]
+		return players[id] as PlayerData
 	else:
 		return null
 
 
 func get_current_player_type() -> int:
-	var curPlayer = get_current_player()
+	var curPlayer := get_current_player()
 	if curPlayer != null:
-		return curPlayer[PLAYER_TYPE]
+		return curPlayer.get_type()
 	else:
 		return -1
 
 
-func update_player(player):
-	var playerId = player[PLAYER_ID]
-	players[playerId] = player
-
+func update_player_from_raw_data(player_data_dictionary: Dictionary):
+	var playerId = player_data_dictionary.id
+	get_player(playerId).load(player_data_dictionary)
 
-func get_host() -> Player:
```

Comment:
lawl bugs

Reply:
Ha, but it worked!  Thanks to strange typing!